sw/airborne/firmwares/rotorcraft/navigation.c

```diff
@@ -298,6 +330,162 @@ bool nav_approaching_from(struct EnuCoor_i *wp, struct EnuCoor_i *from, int16_t
   return false;
 }
 
+/**
+ * @brief function that returns a speed setpoint based on flight plan.
```

Comment:
I don't like the way the navigation code is structured in general. So I'm not sure where to put to keep the code understandable, as it seems to be an alternate nav routine, right ?